### PR TITLE
Add opcode info in compiling-smart-contracts

### LINKED ISSUE #12216: Add info about Opcodes in compiling smart contracts.
In [compiling smart contracts](https://ethereum.org/en/developers/docs/smart-contracts/compiling) page an example of how smart contract is converted to opcodes is provided. But there is missing info about opcodes. 

Adding info about what are opcodes and how it is related to EVM will be helpful.

### PR DESCRIPTION
**Pull request by allwin199**
Compiling smart contract page doesn't have info about opcodes.

## Description

In [compiling smart contracts](https://ethereum.org/en/developers/docs/smart-contracts/compiling) page an example of how smart contract is converted to opcodes is provided. But there is missing info about opcodes. This PR adds info about what are opcodes and what is the relation between opcodes and EVM.

## Related Issue

- Closes #12216

## Summary by CodeRabbit

- **Documentation**
	- Added a section on EVM opcodes in the smart contracts compiling documentation, explaining their significance and providing additional resources for further information.

### CODE REVIEW

File: public/content/developers/docs/smart-contracts/compiling/index.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ contract Greeter {
...
 ```
 
+These are called [Opcodes](https://ethereum.org/developers/docs/evm/opcodes). EVM opcodes are the low-level instructions that the Ethereum Virtual Machine (EVM) can execute. Each opcode represents a specific operation, such as arithmetic operations, logical operations, data manipulation, control flow, etc.
```
Comment (wackerow) [inline public/content/developers/docs/smart-contracts/compiling/index.md:36]: Looks good! Might suggest a tiny change... A pattern we use a lot on the site is to separate a callout link like this after the paragraph, using "More on opcodes". Might suggest this:

```suggestion
These are called **opcodes**. EVM opcodes are the low-level instructions that the Ethereum Virtual Machine (EVM) can execute. Each opcode represents a specific operation, such as arithmetic operations, logical operations, data manipulation, control flow, etc.

[More on opcodes](/developers/docs/evm/opcodes/)
```

(For future reference as well, internal links should only have the relatively path with a trailing slash... ie `/developers/docs/evm/opcodes/`)

**Review by wackerow — CHANGES_REQUESTED** [review]
Thanks @allwin199! Left a suggestion... if that looks good, just drop a comment, or go ahead and commit this suggestion and we'll get it in

File: public/content/developers/docs/smart-contracts/compiling/index.md (hunk 1 above)
Comment (allwin199) [reply public/content/developers/docs/smart-contracts/compiling/index.md:36]: understood! I will follow this suggestion from now on.

Comment (wackerow) [reply public/content/developers/docs/smart-contracts/compiling/index.md]: It's not a pattern that is absolutely required of course, but can be a nice way to let the user read the sentence/paragraph and then be given a link at the end so it doesn't interrupt them mid-sentence.

**Review by wackerow — APPROVED** [review]
Looks good! Thanks @allwin199!

**Comment by wackerow** [comment]
@all-contributors please add @allwin199 for content

**Pull request merged**